### PR TITLE
#539: support advanced ci friendly maven

### PR DESCRIPTION
**Pull request by hohwille**
implement #539

### CODE REVIEW

File: scripts/src/main/resources/scripts/command/mvn [hunk 1]
```diff
@@ -233,25 +244,6 @@ function doCheckTopLevelProject() {
...
-    then
-      doEcho "WARNING: It is not recommended to create releases with SNAPSHOT dependencies!\nThe following dependencies are SNAPSHOT versions:\n${snapshot_list//\[INFO\]/}\nWe strongly recommend to use release versions. Or rerun this command not in batch mode."
-      exit 1
-    else
-      doConfirmWarning "It is not recommended to create releases with SNAPSHOT dependencies!\nThe following dependencies are SNAPSHOT versions:\n${snapshot_list//\[INFO\]/}\nWe strongly recommend to abort and use release versions."
-    fi
-  fi
-  exit 0
-}
-
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/mvn:254]: merging on github failed for whatever reason.

**Review by hohwille — COMMENTED** [review]
Now the diffs look sane and all works as expected.
Providing the revision as argument for CI-friendly continuous `dev-SNAPSHOT` projects is not yet supported.

**Comment by hohwille** [comment]
Forgot shellcheck:
```
Running test #2: test-shellcheck

In scripts/command/mvn line 140:
      updated_config="$(echo "${maven_config}" | sed "s/-Drevision=[^ ]*/-Drevision=${1}/")"
                        ^-- SC2001: See if you can use ${variable//search/replace} instead.

For more information:
  https://www.shellcheck.net/wiki/SC2001
```

**Comment by hohwille** [comment]
```
maven_config="-Drevision=1.2.3 -Dchangelist=-SNAPSHOT"
revision=1.2.3
echo "${maven_config/-Drevision=[^ ]*/-Drevision=${revision}}"
echo $(echo "${maven_config}" | sed "s/-Drevision=[^ ]*/-Drevision=${revision}/")
```
prints this
```
-Drevision=1.2.3
-Drevision=1.2.3 -Dchangelist=-SNAPSHOT
```

So shellcheck is wrong here / false-positive. I will set ignore tags.

**Review by maybeec — APPROVED** [review]

**Pull request merged**